File: Data Structures & Algorithms/longest-increasing-subsequence/submission-3.py
class Solution:
    def lengthOfLIS(self, nums: List[int]) -> int:
        dp = [1 for _ in range(len(nums))]
        max_length = float("-inf")

        for i in range(len(nums) - 1, -1, -1):
            local_max = dp[i]
            for j in range(i + 1, len(nums)):
                if nums[j] > nums[i]:
                    local_max = max(local_max, dp[i] + dp[j])

            dp[i] = local_max
            max_length = max(max_length, local_max)

        return max_length

File: Data Structures & Algorithms/sort-colors/submission-3.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int zeroPos = 0;
        int twoPos = nums.size() - 1;
        int i = 0;

        while(i <= twoPos){
            if (nums[i] == 0){
                nums[i] = nums[zeroPos];
                nums[zeroPos] = 0;
                ++zeroPos;
            }
            else if (nums[i] == 2){
                nums[i] = nums[twoPos];
                nums[twoPos] = 2;
                --twoPos;
                --i;
            }
            ++i;
        }
    }
};